kura/examples/org.eclipse.kura.example.ble.tisensortag.tinyb/src/main/java/org/eclipse/kura/example/ble/tisensortag/tinyb/TiSensorTag.java

```diff
@@ -409,22 +413,39 @@ public void disableAccelerationNotifications() {
      */
     public void setAccelerometerPeriod(int period) {
         byte[] periodBytes = { ByteBuffer.allocate(4).putInt(period).array()[3] };
-        BluetoothLeGattCharacteristic tempPeriodChar;
         try {
             if (this.isCC2650()) {
-                tempPeriodChar = this.gattServices.get(MOVEMENT)
-                        .findCharacteristic(TiSensorTagGatt.UUID_MOV_SENSOR_PERIOD);
-                tempPeriodChar.writeValue(periodBytes);
+                this.gattResources.get(MOVEMENT).getGattService()
+                        .findCharacteristic(TiSensorTagGatt.UUID_MOV_SENSOR_PERIOD).writeValue(periodBytes);
             } else {
-                tempPeriodChar = this.gattServices.get(ACCELEROMETER)
-                        .findCharacteristic(TiSensorTagGatt.UUID_ACC_SENSOR_PERIOD);
-                tempPeriodChar.writeValue(periodBytes);
+                this.gattResources.get(ACCELEROMETER).getGattService()
+                        .findCharacteristic(TiSensorTagGatt.UUID_ACC_SENSOR_PERIOD).writeValue(periodBytes);
             }
         } catch (KuraException e) {
             logger.error("Acceleration period set failed", e);
         }
     }
 
+    public boolean isAccelerometerNotifying() {
+        TiSensorTagGattResources resource;
+        if (this.isCC2650()) {
+            resource = this.gattResources.get(MOVEMENT);
+            if (resource != null) {
```

Comment:
Since it's the same in both branches of the if, I believe it can be put outside of the if